sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncClient.java

```diff
@@ -240,19 +234,8 @@ boolean isContentResponseOnWriteEnabled() {
      * @return the mono.
      */
     public Mono<CosmosDatabaseResponse> createDatabaseIfNotExists(String id, ThroughputProperties throughputProperties) {
```

Comment:
We could do that, but avoiding nesting call in public api. Also it would be easier to read and debug. And our tracer goal getting fulfill with current design, so if you don't mind, can we keep it in current state?

Reply:
same here, tracking the work item here: https://github.com/Azure/azure-sdk-for-java/issues/13031